app/controllers/users_controller.rb

```diff
@@ -0,0 +1,7 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by!(username: params.fetch(:id))
```

Comment:
While params.fetch(:id) is perfectly valid for ensuring the presence of a parameter, its use here might be overkill unless you have a specific need to handle the absence of :id differently than how Rails handles missing parameters for route-based actions. If you're not performing any special error handling for this case, params[:id] would suffice and is more idiomatic in Rails controllers.